tasks/common.yml

```diff
@@ -42,40 +16,75 @@
     groups: "audio"
     home: "/var/lib/archivematica"
 
-- name: "Add archivematica user permissions in visudo file"
-  lineinfile:
-    dest: "/etc/sudoers"
-    state: "present"
-    regexp: "^archivematica ALL\\="
-    line: "archivematica ALL=NOPASSWD:/bin/mv,/bin/chown,/bin/chmod,/usr/bin/find,/usr/bin/gs,/usr/bin/inkscape"
-    validate: "/usr/sbin/visudo -cf %s"
-  when: archivematica_src_enable_sudoers is defined and archivematica_src_enable_sudoers|bool
-
-- name: "Remove archivematica user permissions in visudo file"
-  lineinfile:
-    dest: "/etc/sudoers"
-    state: "absent"
-    regexp: "^archivematica ALL\\="
-    line: "archivematica ALL=NOPASSWD:/bin/mv,/bin/chown,/bin/chmod,/usr/bin/find,/usr/bin/gs,/usr/bin/inkscape"
-    validate: "/usr/sbin/visudo -cf %s"
-  when: archivematica_src_enable_sudoers is not defined or not archivematica_src_enable_sudoers|bool
-
+#
+# Prepare pip
+#
+- name: "Common configuration for source installs"
+  block:
+  - name: "Ensure pip is not installed from packages"
+    package:
+      name: "{{ item }}"
+      state: "absent"
+    with_items:
+      - "python-pip"
+      - "python2-pip"
+
+  - name: "Download get-pip.py"
+    get_url:
+      url: "https://bootstrap.pypa.io/get-pip.py"
+      dest: "/root/get-pip.py"
+
+  - name: "Install pip with get-pip.py"
+    command: "python get-pip.py"
+    args:
+      chdir: "/root/"
+
+  - name: "Install virtualenv with pip"
+    pip:
+      name: "{{ item }}"
+      state: "latest"
+    with_items:
+      - "virtualenv"
 #
 # Prepare `archivematica_src_dir`
 #
 
-- name: "Expand archivematica_src_dir"
-  set_fact:
-    archivematica_src_dir: "{{ archivematica_src_dir|expanduser }}"
+  - name: "Create archivematica_src_dir"
+    file:
+      state: "directory"
+      path: "{{ archivematica_src_dir }}"
+  when:
+    - (archivematica_src_install_am|bool or archivematica_src_install_ss|bool)
+
+#
+# Configure RPM repositories and dependencies
+#
 
-- name: "Create archivematica_src_dir"
-  file:
-    state: "directory"
-    path: "{{ archivematica_src_dir }}"
+- name: "Configure rpm-install dependencies"
+  block:
+
+  - name: "Install necessary packages required by rpms"
+    yum:
+      name: "{{ item }}"
+      state: "present"
+    with_items:
+      - "python-virtualenv"
+      - "python-pip"
+
+  - name: "Add AM repos"
+    yum_repository:
+      name: "{{ item.key }}"
+      description: "{{ item.key }}"
+      baseurl: "{{ item.value.baseurl }}"
+      repo_gpgcheck: "{{ item.value.gpgcheck }}"
+    loop: "{{ lookup('dict',archivematica_src_rpm_repositories) }}"
```

Comment:
The repo needs the gpgkey url, I got the error:

```
TASK [external-roles/artefactual-archivematica-qa-src : Install necessary packages required by this ansible role] *************************************************************************************************
[DEPRECATION WARNING]: Invoking "yum" only once while using a loop via squash_actions is deprecated. Instead of using a loop to supply multiple items and specifying `name: "{{ item }}"`, please use `name: 
['libsemanage-python', 'policycoreutils-python']` and remove the loop. This feature will be removed in version 2.11. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
failed: [mamedin-ansible-rpms] (item=[u'libsemanage-python', u'policycoreutils-python']) => {"ansible_loop_var": "item", "changed": false, "item": ["libsemanage-python", "policycoreutils-python"], "msg": "https://packages.archivematica.org/1.10.x/centos/repodata/repomd.xml: [Errno -1] Gpg Keys not imported, cannot verify repomd.xml for repo archivematica\nTrying other mirror.\n\n\n One of the configured repositories failed (archivematica),\n and yum doesn't have enough cached data to continue. At this point the only\n safe thing yum can do is fail. There are a few ways to work \"fix\" this:\n\n     1. Contact the upstream for the repository and get them to fix the problem.\n\n     2. Reconfigure the baseurl/etc. for the repository, to point to a working\n        upstream. This is most often useful if you are using a newer\n        distribution release than is supported by the repository (and the\n        packages for the previous distribution release still work).\n\n     3. Run the command with the repository temporarily disabled\n            yum --disablerepo=archivematica ...\n\n     4. Disable the repository permanently, so yum won't use it by default. Yum\n        will then just ignore the repository until you permanently enable it\n        again or use --enablerepo for temporary usage:\n\n            yum-config-manager --disable archivematica\n        or\n            subscription-manager repos --disable=archivematica\n\n     5. Configure the failing repository to be skipped, if it is unavailable.\n        Note that yum will try to contact the repo. when it runs most commands,\n        so will have to try and fail each time (and thus. yum will be be much\n        slower). If it is a very temporary problem though, this is often a nice\n        compromise:\n\n            yum-config-manager --save --setopt=archivematica.skip_if_unavailable=true\n\nfailure: repodata/repomd.xml from archivematica: [Errno 256] No more mirrors to try.\nhttps://packages.archivematica.org/1.10.x/centos/repodata/repomd.xml: [Errno -1] Gpg Keys not imported, cannot verify repomd.xml for repo archivematica\n", "rc": 1, "results": []}
```
When running `sudo yum repolist` on CentOS VM:

```
https://packages.archivematica.org/1.10.x/centos/repodata/repomd.xml: [Errno -1] Gpg Keys not imported, cannot verify repomd.xml for repo archivematica
...
https://packages.archivematica.org/1.10.x/centos-extras/repodata/repomd.xml: [Errno -1] Gpg Keys not imported, cannot verify repomd.xml for repo archivematica-extras
``` 
Taking a look to the archivematica-docs CentOS install script ( https://raw.githubusercontent.com/artefactual/archivematica-docs/1.10/admin-manual/installation-setup/installation/scripts/am-centos-rpm.sh ) this task needs to add to both yum repo files:

```
gpgkey=https://packages.archivematica.org/1.10.x/key.asc
```

Reply:
Added gpgkey and gpgcheck to the configuration dict.